driver-core/src/main/java/com/datastax/driver/core/InboundTrafficMeter.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright DataStax, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.datastax.driver.core;
+
+import com.codahale.metrics.Meter;
+import io.netty.buffer.ByteBuf;
+import io.netty.channel.ChannelHandler.Sharable;
+import io.netty.channel.ChannelHandlerContext;
+import io.netty.channel.ChannelInboundHandlerAdapter;
+
+@Sharable
+class InboundTrafficMeter extends ChannelInboundHandlerAdapter {
+
+    private final Meter meter;
+
+    InboundTrafficMeter(Meter meter) {
+        this.meter = meter;
+    }
+
+    @Override
+    public void channelRead(ChannelHandlerContext ctx, Object msg) throws Exception {
+        if (msg instanceof ByteBuf) {
+            meter.mark(((ByteBuf) msg).readableBytes());
+        }
+        super.channelRead(ctx, msg);
+    }
```

Comment:
Nice, straightforward way to implement this 💯 

I was wondering if we could use a single `ChannelDuplexHandler`, but the two methods don't share any state so I it doesn't really make any difference.